### PR TITLE
Port HIG site example-app-base changes

### PR DESCRIPTION
**Pull request by jordandrako**
#### Pull request checklist

- [x] Include a change request file using `$ npm run change`

#### Description of changes

These are changes I've made that are required by the new Fabric HIG site. I need these merged upstream to make pulling upstream changes easier.

**EditSection:**

- Allow `className` to be passed to root.
- Allow content to be more flexible, while still only rendering if it finds a `PageMarkdown` component (either as the direct child or within the first level of the array of children).

**MarkdownTable:**

- Created `MarkdownTable` components intended to be consumed by `PageMarkdown` so tables are able to be correctly.
- Includes components for: table, thead, tbody, tr, and cell which covers th and td.

**PageMarkdown:**

- Add global `className`s to overridden tags.
- Override table tags with new `MarkdownTable` components.
- Give `PageMarkdown` a static `displayName` so `EditSection` can find if the content contains specifically a `PageMarkdown` component in order to render. This make using `PageMarkdown` in content a LOT more flexible.
- Give root a global `className`.

#### Focus areas to test

See if markdown content and edit buttons still render correctly on fabric-website. From my checks it does.

### CODE REVIEW

**Comment by ecraig12345** [comment]
@JasonGore @mikewheaton Based on recent Git history I'm guessing that one of you might be better able to review this than I am? Or feel free to suggest someone else.

File: packages/example-app-base/src/components/MarkdownTable/MarkdownTable.module.scss [hunk 1]
```diff
@@ -0,0 +1,59 @@
+@import '../../common/common';
+
+.Table {
```
Comment (ecraig12345) [inline packages/example-app-base/src/components/MarkdownTable/MarkdownTable.module.scss]: These class names should probably be lowercase to match everywhere else.

File: packages/example-app-base/src/components/MarkdownTable/MarkdownTable.module.scss [hunk 2]
```diff
@@ -0,0 +1,59 @@
+@import '../../common/common';
```
Comment (ecraig12345) [inline packages/example-app-base/src/components/MarkdownTable/MarkdownTable.module.scss]: Would it be possible to use css-in-js here instead? This is a lot of scss to be adding in a brand new file when we're trying to move away from scss.

Comment (jordandrako) [reply packages/example-app-base/src/components/MarkdownTable/MarkdownTable.module.scss]: I'll see if I can dedicate some time to do css-in-js instead.
Also tagging @Jahnp for visibility.

Comment (ecraig12345) [reply packages/example-app-base/src/components/MarkdownTable/MarkdownTable.module.scss]: For a new file, it's not terribly difficult. I've just recently been learning about it while working on the DetailsList examples in #7464. Here's an example of part of the conversion...though what you're doing is a bit more complicated and involves some things I'm not quite sure how to do yet (some guesses included below). @Vitalius1 would probably have a better idea of how to do complicated selectors with all the css-in-js component conversions he's been doing.

```ts
import { mergeStyleSets, DefaultPalette, DefaultFontStyles } from 'office-ui-fabric-react/lib/Styling';

// This generates a map of class names
const classNames1 = mergeStyleSets({
  table: {
    width: '100%',
    borderCollapse: 'collapse',
    // compiler complained about borderSpacing--maybe you meant cellSpacing?
    cellSpacing: 0,
    selectors: {
      '.ms-Image img': {
        maxWidth: '100%'
      }
    }
  },
  th: {
    fontWeight: 600,
    // Not sure if this is the correct way to get the colors...
    color: DefaultPalette.neutralPrimary,
    whiteSpace: 'nowrap',
    textAlign: 'left'

  },
  // ...

  // Not 100% sure how to handle nested selectors...my ideas:
  // - make a separate class and apply both it and the regular class in the JS file?
  // trInTh: { /*styles*/ }
});
// - make a second style object and build the nested selector from the generated class names?
// const finalClassNames = mergeStyleSets(classNames1, {
//   [`.${classNames1.th} .${classNames1.tr}`]: { /*styles */ }
// });
// - or some other way entirely?
```

Comment (ecraig12345) [reply packages/example-app-base/src/components/MarkdownTable/MarkdownTable.module.scss]: Looks good as far as I can tell. But I'd really like for someone else who's familiar with example-app-base to review the changes.

**Comment by jordandrako** [comment]
@kenotron I've tested that all the current docs render correctly still. I couldn't find and instances where there were tables in the current docs.

File: packages/example-app-base/src/components/EditSection/EditSection.tsx [hunk 3]
```diff
@@ -1,36 +1,87 @@
...
-          aria-labelledby={`${title}-${section}-editButtonHost`}
-          iconProps={{ iconName: 'Edit' }}
-          href={url}
-          target="_blank"
-          rel="noopener noreferrer"
-        />
-      </TooltipHost>
+      this._shouldRender && (
+        <div className={className}>
+          <TooltipHost
```
[the remark this answers is not in the document]
Comment (jordandrako) [reply packages/example-app-base/src/components/EditSection/EditSection.tsx]: At the time I wrote these for the HIG site, TooltipHost didn't pass the `className`. I see now it accepts `hostClassName`, I can adjust that.

**Pull request merged**